filenamify.js

```diff
@@ -21,6 +21,26 @@ const filenamify = (string, options = {}) => {
 		throw new Error('Replacement string cannot contain reserved filename characters');
 	}
 
+	if (string.includes(replacement)) {
+		const substrings = string.split(replacement);
+		const tmp = [];
+		substrings.forEach((substring, substringIndex) => {
+			let filenamified = '';
+			if (substring.length > 0) {
+				filenamified = filenamify(substring, options);
+				if (substringIndex === substrings.length - 1 || substringIndex === 0) {
+					filenamified = stripOuter(filenamified, replacement);
+				} else if (filenamified.length === 0) {
+					filenamified = replacement;
+				}
+			}
+
+			tmp.push(filenamified);
+		});
+
+		return tmp.join(replacement);
```

Comment:
All the logic in this scope could be moved out into a function.

Reply:
Got it. I've moved it to the `recursivelyFilenamify` function at the end of this file.